src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -206,9 +217,10 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
                         interactive={canEditAmount}
                         shouldShowRightIcon={canEditAmount}
                         onPress={() => Navigation.navigate(ROUTES.EDIT_REQUEST.getRoute(report.reportID, CONST.EDIT_REQUEST_FIELD.AMOUNT))}
-                        brickRoadIndicator={hasErrors && transactionAmount === 0 ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                        brickRoadIndicator={hasViolations('amount') || (hasErrors && transactionAmount === 0) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
                         error={hasErrors && transactionAmount === 0 ? translate('common.error.enterAmount') : ''}
                     />
+                    {canUseViolations && <FieldViolationMessages violations={getViolationsForField('amount')} />}
```

Comment:
NAB: Do we have const/types we can use instead of the string literals for fields like `amount`?

Reply:
We can't use types at runtime. Best practice in Typescript is to use string literals because they are type safe and autocomplete will suggest the right answer. They are essentially the same as an enum in both function and DX.

We discussed this in the previous thread about sting literals.

Reply:
Awesome, thanks for that info, I'm learning quite a bit about TS best practices 🙇

Reply:
Here's an example of the type of error you get if you try to use a value outside of the union:

![CleanShot 2023-12-12 at 12 11 28@2x](https://github.com/Expensify/App/assets/22041394/1ddeb66b-4819-42eb-b582-c0489b4a78b8)

Reply:
And autocomplete only suggests values compatible with the type.

![CleanShot 2023-12-12 at 12 12 44@2x](https://github.com/Expensify/App/assets/22041394/12ea1865-777d-4e75-abd4-fd193754ccc6)